pipeline/splicing/calcVarPriors.py

```diff
@@ -935,79 +1054,130 @@ def isCIDomainInRegion(regionStart, regionEnd, boundaries, gene):
+        else:
+            start = regionStart
+            end = regionEnd
         for domain in brca1CIDomains[boundaries].keys():
             domainStart = brca1CIDomains[boundaries][domain]["domStart"]
             domainEnd = brca1CIDomains[boundaries][domain]["domEnd"]
-            overlap = range(max(regionEnd, domainEnd), min(regionStart, domainStart) + 1)
+            overlap = range(max(end, domainEnd), min(start, domainStart) + 1)
             if len(overlap) > 0:
                 return True
     elif gene == "BRCA2":
+        if regionStart < regionEnd:
+            start = regionStart
+            end = regionEnd
+        else:
+            start = regionEnd
+            end = regionStart
         for domain in brca2CIDomains[boundaries].keys():
             domainStart = brca2CIDomains[boundaries][domain]["domStart"]
             domainEnd = brca2CIDomains[boundaries][domain]["domEnd"]
-            overlap = range(max(regionStart, domainStart), min(regionEnd, domainEnd) + 1)
+            overlap = range(max(start, domainStart), min(end, domainEnd) + 1)
             if len(overlap) > 0:
                 return True
     return False
 
-def getRefExonLength(variant):
-    '''Given a variant, returns the length of the reference exon'''
+def getRefExonLength(variant, donor=True):
+    '''
+    Given a variant, returns the length of the reference exon
+    If variant is in an exon, returns length of that exon
+    If variant is in a reference splice region, returns length of exon in which exonic portion is included
+    If variant is in intron, returns exon in which either closest splice donor or acceptor is included depending on donor argument
+      If donor=True, returns exon length for previous exon
+      If donor=False, returns exon length for subsequent exon
+    '''
     if varInExon(variant) == True:
         varExonNum = getVarExonNumberSNS(variant)
-        exonBounds = getExonBoundaries(variant)
-        if getVarStrand(variant) == "-":
-            varExonStart = int(exonBounds[varExonNum]["exonStart"])
-            varExonEnd = int(exonBounds[varExonNum]["exonEnd"])
-            exonLength = varExonStart - varExonEnd
+    else:
+        if varInSpliceRegion(variant, donor=donor, deNovo=False) == True:
+            spliceBounds = getVarSpliceRegionBounds(variant, donor=donor, deNovo=False)
+            varExonNum = spliceBounds["exonName"]
         else:
-            varExonStart = int(exonBounds[varExonNum]["exonStart"])
-            varExonEnd = int(exonBounds[varExonNum]["exonEnd"])
-            exonLength = varExonEnd - varExonStart
-        return exonLength
+            varExonNum = getClosestExonNumberIntronicSNS(variant, "enigma", donor=donor)
+    exonBounds = getExonBoundaries(variant)
+    if getVarStrand(variant) == "-":
+        varExonStart = int(exonBounds[varExonNum]["exonStart"])
+        varExonEnd = int(exonBounds[varExonNum]["exonEnd"])
+        exonLength = varExonStart - varExonEnd
```

Comment:
The +1 is kind of hidden because of the RefSeq numbering (which starts to the right of the first base). For BRCA1 the exonEnd is 1 bp too short and for BRCA2 the exonStart is 1 bp too long so it works out. Do you want me to add a comment to explain this?

Reply:
yes please!